servant-js/src/Servant/JS/Angular.hs

```diff
@@ -0,0 +1,139 @@
+module Servant.JS.Angular where
+
+import Servant.JS.Internal
+import Control.Lens
+import Data.List
+import Data.Monoid
+
+-- | Options specific to the angular code generator
+data AngularOptions = AngularOptions
+  { serviceName :: String                         -- ^ When generating code with wrapInService,
+                                                  --   name of the service to generate
+  , prologue :: String -> String -> String        -- ^ beginning of the service definition
+  , epilogue :: String                            -- ^ end of the service definition
+  }
+
+-- | Default options for the Angular codegen. Used by 'wrapInService'.
+defAngularOptions :: AngularOptions
+defAngularOptions = AngularOptions
+  { serviceName = ""
+  , prologue = \svc m -> m <> "service('" <> svc <> "', function($http) {\n"
+                           <> "  return ({"
+  , epilogue = "});\n});\n"
+    }
+
+-- | Instead of simply generating top level functions, generates a service instance
+-- on which your controllers can depend to access your API.
+-- This variant uses default 'AngularOptions'.
+angularService :: AngularOptions -> JavaScriptGenerator
+angularService ngOpts = angularServiceWith ngOpts defCommonGeneratorOptions
+
+-- | Instead of simply generating top level functions, generates a service instance
+-- on which your controllers can depend to access your API
+angularServiceWith :: AngularOptions -> CommonGeneratorOptions -> JavaScriptGenerator
+angularServiceWith ngOpts opts reqs = 
+    prologue ngOpts svc mName
+    <> intercalate "," (map generator reqs) <>
+    epilogue ngOpts
+    where
+        generator req = generateAngularJSWith ngOpts opts req
+        svc = serviceName ngOpts
+        mName = if null (moduleName opts)
+                   then "app."
+                   else moduleName opts <> "."
+
+-- | Generate regular javacript functions that use
+--   the $http service, using default values for 'CommonGeneratorOptions'.
+angular :: AngularOptions -> JavaScriptGenerator
+angular ngopts = angularWith ngopts defCommonGeneratorOptions
+
+-- | Generate regular javascript functions that use the $http service.
+angularWith :: AngularOptions -> CommonGeneratorOptions -> JavaScriptGenerator
+angularWith ngopts opts = intercalate "\n\n" . map (generateAngularJSWith ngopts opts)
+
+-- | js codegen using $http service from Angular using default options
+generateAngularJS :: AngularOptions -> AjaxReq -> String
+generateAngularJS ngOpts = generateAngularJSWith ngOpts defCommonGeneratorOptions
+    
+-- | js codegen using $http service from Angular
```

Comment:
in angular, they call their "modules" $http, $service, etc...
(js guys like the $ sign)